src/main/app-updater.ts

```diff
@@ -1,19 +1,170 @@
+        cleanupChannels();
+
+        await autoUpdater.downloadUpdate();
+        autoUpdater.quitAndInstall();
+
+        resolve();
+      })
+      .on(yesLaterChannel, async () => {
+        cleanupChannels();
+
+        await autoUpdater.downloadUpdate();
+        autoUpdater.autoInstallOnAppQuit = true;
+
+        resolve();
+      })
+      .on(noChannel, () => {
+        cleanupChannels();
+        resolve();
+      });
+
+    broadcastIpc({
+      channel: NotificationChannelAdd,
+      args: [{
+        title,
+        body,
+        status: "info",
+        buttons: [
+          {
+            label: "Yes, now",
+            backchannel: yesNowChannel,
+            style: {
+              background: "green",
+              marginRight: "10px"
+            }
+          },
+          {
+            label: "Yes, on quit",
+            backchannel: yesLaterChannel,
+            style: {
+              background: "green",
+              marginRight: "10px"
+            }
+          },
+          {
+            label: "No",
+            backchannel: noChannel,
+            accent: true
+          }
+        ],
+        closeChannel: noChannel,
+      }]
+    });
+  });
+}
+
+/**
+ * starts the automatic update checking
+ * @param interval milliseconds between interval to check on, defaulkts to 24h
+ */
+export function startUpdateChecking(interval = 1000 * 60 * 60 * 24): void {
+  if (isDevelopment) {
+    return;
   }
 
-  public start() {
-    setInterval(() => {
-      autoUpdater.checkForUpdatesAndNotify()
-    }, this.updateInterval)
+  autoUpdater.logger = logger;
+
+  autoUpdater
+    .on("update-available", async (args: UpdateInfo) => {
+      try {
+        const releaseDate = new Date(args.releaseDate);
+        const body = `Version ${args.version} was release on ${dateFormat(releaseDate, "dddd, mmmm dS, yyyy")}.`;
+        broadcastIpc({
+          channel: NotificationChannelAdd,
+          args: [{
+            title,
+            body,
+            status: "info",
+            timeout: 5000,
+          }]
+        });
 
-    return autoUpdater.checkForUpdatesAndNotify()
+        await autoUpdateCheck(args);
+      } catch (error) {
+        logger.error("[UPDATE CHECKER]: notification failed", { error: String(error) })
+      }
+    })
+    .on("update-not-available", (args: UpdateInfo) => {
+      try {
+        const version = new SemVer(args.version);
+        const stream = version.prerelease !== null ? "prerelease" : "stable";
+        const body = `Lens is running the latest ${stream} version.`;
+        broadcastIpc({
+          channel: NotificationChannelAdd,
+          args: [{
+            title,
+            body,
+            status: "info",
+            timeout: 5000,
+          }]
+        })
+      } catch (error) {
+        logger.error("[UPDATE CHECKER]: notification failed", { error: String(error) })
+      }
+    })
+
+  async function helper() {
+    while (true) {
```

Comment:
`while(true){}` always looks scary :D Why this is needed?

Reply:
I don't want to use `setTimeout` as that doesn't allow for catching errors (easily, we could use promises again I guess). This is essentially the same as calling `setTimeout` multiple times but it is more explicit. The `delay` function makes this "long polling".

Reply:
`while(true){}` is only scary when you expect execution to eventually break out of the loop